### DISCUSSION TITLE
Locking full Livewire components

### DISCUSSION THREAD

**Discussion by chillbram** [livewire/livewire#8327] [3. Feature ideas]
I think it would be great to able to apply `#[Locked]` on a Livewire component, which would lock all public properties at once rather than having to define `#[Locked]` for every property. It can lead to a lot of visual noise and when using other attributes like `#[Validate]` it's less clear at a glance what can and can't be changed from the front-end.

I've got some Livewire components where all or a large majority of properties should not be changed by users directly and interaction is based on method calls. To facilitate more use cases an `#[Unlocked]` attribute could then allow individual properties to be open to change. Locking a class and then just unlocking a single one makes it clearer where the security dangers lie, as one can easily look past one non-locked property sitting in a sea of locked attributes.

**Comment by lrljoe** [comment]
Use Computed Properties if you just want to pull values to the frontend.

That's much more efficient, and the entire purpose of the feature.

Not sure why that wouldn't be the go-to?  Particularly when the caching benefits etc kick in

There is a PR for a global behaviour like this tho: #8586

**Comment by chillbram** [comment]
Computed properties are indeed often the solution for read-only data, but they don't allow for user changed data (through methods in this case, rather than as a directly modeled property). Practically all button interactions in my applications are coupled to some way to `#[Locked]` properties, which is great as it allows for complete control of possible values they can contain. 

I hope something great comes from that PR :)
